### PR TITLE
Add NSX-T IPsec VPN Tunnel support

### PR DESCRIPTION
**Pull request by Didainius**
As part of https://github.com/vmware/terraform-provider-vcd/issues/618. this PR adds `NsxtIpSecVpnTunnel` and `types.NsxtIpSecVpnTunnel` types to manage IPsec VPN Tunnels on NSXT edge gateways.

The following methods are created:
* `NsxtEdgeGateway.GetAllIpSecVpnTunnels` 
* `NsxtEdgeGateway.GetIpSecVpnTunnelById`
* `NsxtEdgeGateway.GetIpSecVpnTunnelByName`
* `NsxtEdgeGateway.CreateIpSecVpnTunnel`
* `NsxtIpSecVpnTunnel.Update`
* `NsxtIpSecVpnTunnel.Delete`
* `NsxtIpSecVpnTunnel.GetStatus`
* `NsxtIpSecVpnTunnel.UpdateTunnelConnectionProperties`
* `NsxtIpSecVpnTunnel.GetTunnelConnectionProperties`
* `NsxtIpSecVpnTunnel.IsEqualTo`

As well as more new types:
* `types.NsxtIpSecVpnTunnel`
* `types.NsxtIpSecVpnTunnelStatus`
* `types.NsxtIpSecVpnTunnelSecurityProfile`

### CODE REVIEW

File: govcd/nsxt_ipsec_vpn_tunnel.go [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+	urlRef, err := ipSecVpn.client.OpenApiBuildEndpoint(fmt.Sprintf(endpoint, ipSecVpn.edgeGatewayId, ipSecVpn.NsxtIpSecVpn.ID))
+	if err != nil {
+		return nil, err
+	}
+
+	ipSecVpnTunnelStatus := &types.NsxtIpSecVpnTunnelStatus{}
+
+	err = ipSecVpn.client.OpenApiGetItem(minimumApiVersion, urlRef, nil, ipSecVpnTunnelStatus, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error deleting NSX-T IPsec VPN Tunnel configuration: %s", err)
```
Comment (dataclouder) [inline govcd/nsxt_ipsec_vpn_tunnel.go:257]:
```suggestion
		return nil, fmt.Errorf("error getting NSX-T IPsec VPN Tunnel status: %s", err)
```

File: govcd/nsxt_ipsec_vpn_tunnel.go [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+	urlRef, err := ipSecVpn.client.OpenApiBuildEndpoint(fmt.Sprintf(endpoint, ipSecVpn.edgeGatewayId, ipSecVpn.NsxtIpSecVpn.ID))
+	if err != nil {
+		return nil, err
+	}
+
+	ipSecVpnTunnelProfile := &types.NsxtIpSecVpnTunnelSecurityProfile{}
+	err = ipSecVpn.client.OpenApiGetItem(minimumApiVersion, urlRef, nil, ipSecVpnTunnelProfile, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error removing NSX-T IPsec VPN Connection Properties: %s", err)
```
Comment (dataclouder) [inline govcd/nsxt_ipsec_vpn_tunnel.go:312]:
```suggestion
		return nil, fmt.Errorf("error retrieving NSX-T IPsec VPN Connection Properties: %s", err)
```

**Review by dataclouder — CHANGES_REQUESTED** [review]
minor changes needed

File: govcd/nsxt_ipsec_vpn_tunnel.go (hunk 1 above)
Comment (Didainius) [reply govcd/nsxt_ipsec_vpn_tunnel.go]: Fixed.

File: govcd/nsxt_ipsec_vpn_tunnel.go (hunk 2 above)
Comment (Didainius) [reply govcd/nsxt_ipsec_vpn_tunnel.go]: Fixed.

**Review by dataclouder — APPROVED** [review]

**Pull request merged**